src/server/checkBrowserVersion.js

```diff
@@ -0,0 +1,37 @@
+const childProcess = require('child_process');
+
+const firefox = require('selenium-webdriver/firefox');
+
+const config = require('./config');
+
+const MINIMUM_FIREFOX_VERSION = 50.0;
+const FIREFOX_VERSION_MATCHER = /Mozilla Firefox ([0-9.]+)/;
+
+module.exports = function checkBrowserVersion() {
+  if (config.driver !== 'firefox') {
+    // Our main browser target is Firefox. If you are using something else, you
+    // must know what you are doing. :)
+    return Promise.resolve();
+  }
+
+  return new Promise((resolve, reject) => {
+    new firefox.Binary().locate().then((pathToExecutable) => {
+      childProcess.exec(`${pathToExecutable} --version`, (error, stdout) => {
+        if (error) {
+          reject(new Error(`Failed to check Firefox version: ${error}`));
+          return;
+        }
+        const match = stdout.match(FIREFOX_VERSION_MATCHER);
+        if (!match) {
+          reject(new Error(`Failed to parse Firefox version string "${stdout}"`));
+        } else if (parseFloat(match[1]) < MINIMUM_FIREFOX_VERSION) {
```

Comment:
I am currently running Firefox 47.0.1. I think we should probably use something that can compare semver numbers to each other instead of just parseFloat.